cmake/IRODSConfigVersion.cmake.in

```diff
@@ -0,0 +1,158 @@
+          # looking for 4.3 or 4.9x, we are 4.9x
+          set(PACKAGE_VERSION_COMPATIBLE TRUE)
+        else()
+          set(PACKAGE_VERSION_COMPATIBLE FALSE)
+        endif()
+      else()
+        set(PACKAGE_VERSION_COMPATIBLE TRUE)
+      endif()
+    endif()
+  elseif (IRODS_VERSION_3 GREATER_EQUAL "90")
+    # development minor version
+    math(EXPR IRODS_VERSION_2_NEXT "${IRODS_VERSION_2} + 1")
+    set(IRODS_DEVELOPMENT_VERSION_FOR "${IRODS_VERSION_1}.${IRODS_VERSION_2_NEXT}.0")
+    if (IRODS_DEVELOPMENT_VERSION_FOR VERSION_EQUAL PACKAGE_FIND_VERSION)
+      # looking for next minor release
+      set(PACKAGE_VERSION_COMPATIBLE TRUE)
+    elseif(
+      IRODS_VERSION_1 EQUAL PACKAGE_FIND_VERSION_MAJOR
+        AND PACKAGE_VERSION VERSION_GREATER_EQUAL PACKAGE_FIND_VERSION
+    )
+      # same major release, looking for equal or lesser
+      set(PACKAGE_VERSION_COMPATIBLE TRUE)
+    else()
+      set(PACKAGE_VERSION_COMPATIBLE FALSE)
+    endif()
+  elseif (IRODS_VERSION_1 GREATER_EQUAL "5")
+    # not a development version
+    # major release is first part of version number
+    if (
+      IRODS_VERSION_1 EQUAL PACKAGE_FIND_VERSION_MAJOR
+        AND PACKAGE_VERSION VERSION_GREATER_EQUAL PACKAGE_FIND_VERSION
+    )
+      # major release matches, looking for equal or lesser
+      # covers case of looking for development version for next major release
+      set(PACKAGE_VERSION_COMPATIBLE TRUE)
+    elseif (PACKAGE_FIND_VERSION_MINOR GREATER_EQUAL "90")
+      # looking for development major version
+      math(EXPR PACKAGE_FIND_VERSION_1_NEXT "${PACKAGE_FIND_VERSION_MAJOR} + 1")
+      if (PACKAGE_FIND_VERSION_1_NEXT EQUAL IRODS_VERSION_1)
+        # looking for development version leading up to current major release - compatible
+        set(PACKAGE_VERSION_COMPATIBLE TRUE)
+      else()
+        set(PACKAGE_VERSION_COMPATIBLE FALSE)
+      endif()
+    else()
+      set(PACKAGE_VERSION_COMPATIBLE FALSE)
+    endif()
+  elseif (
+    IRODS_VERSION_1 EQUAL PACKAGE_FIND_VERSION_MAJOR
+      AND IRODS_VERSION_2 EQUAL PACKAGE_FIND_VERSION_MINOR
+      AND PACKAGE_VERSION VERSION_GREATER_EQUAL PACKAGE_FIND_VERSION
+  )
+    # major release is second part of version number
+    # development versions not a consideration here
+    # major release matches, looking for equal or lesser
+    set(PACKAGE_VERSION_COMPATIBLE TRUE)
+  else()
+    set(PACKAGE_VERSION_COMPATIBLE FALSE)
+  endif()
+endif()
+
+# After checking version number, check bit-ness
+set(IRODS_SIZEOF_VOID_P "@CMAKE_SIZEOF_VOID_P@")
+
+# if the installed or the calling project don't have CMAKE_SIZEOF_VOID_P set, ignore it:
+if ("${CMAKE_SIZEOF_VOID_P}" STREQUAL "" OR "${IRODS_SIZEOF_VOID_P}" STREQUAL "")
+  return()
+endif()
+
+if (NOT CMAKE_SIZEOF_VOID_P STREQUAL IRODS_SIZEOF_VOID_P)
+  math(EXPR installedBits "${IRODS_SIZEOF_VOID_P} * 8")
+  set(PACKAGE_VERSION "${PACKAGE_VERSION} (${installedBits}bit)")
+  set(PACKAGE_VERSION_UNSUITABLE TRUE)
+endif()
```

Comment:
What is this attempting to solve?

Is it about detecting whether a platform is 64-bit or something?

Reply:
Yeah, it's to make sure the bit-ness matches. If you're building a 32-bit project, you don't want to link against a 64-bit library. This is standard for CMake package version files, and is already present in the ones we ship today.

Reply:
Yep, that makes sense.

Reply:
What happens to the consuming project if L151 is executed?

Reply:
> What happens to the consuming project if L151 is executed?

Here, `return()` is equivalent to jumping to the end of the file.

Reply:
I should've been clearer. The question is ... if L151 is reached, that means the size of a void pointer could not be detected.

What does that mean for the consuming project?
Does it matter at all?

Reply:
If the bit-ness test can't be run due to lack of `CMAKE_SIZEOF_VOID_P`, it's up to the person calling `cmake` to make sure that the iRODS installation found is compatible in this way, and if not, to define CMake variables on the command line to point CMake in the right direction, just like for any other incompatibility. It's not the responsibility of the consuming project to manage this.

Reply:
Works for me.